src/backend/joanie/core/serializers.py

```diff
@@ -1,71 +1,271 @@
+        """
+        return CourseRunSerializer(
+            target_course.course_runs.order_by("start"), many=True
+        ).data
+
+
+class ProductSerializer(serializers.ModelSerializer):
+    """
+    Product serializer including
+        - certificate information if there is
+        - targeted courses with its course runs
+            - If user is authenticated, we try to retrieve enrollment related
+              to each course runs.
+        - order if user is authenticated
+    """
+
+    id = serializers.CharField(source="uid", read_only=True)
+    certificate = CertificationDefinitionSerializer(
+        read_only=True, source="certificate_definition"
+    )
+    currency = serializers.SerializerMethodField()
+    price = serializers.DecimalField(
+        max_digits=9,
+        decimal_places=2,
+        coerce_to_string=False,
+        min_value=0,
+    )
+    target_courses = serializers.SerializerMethodField()
+
+    class Meta:
+        model = models.Product
+        fields = [
+            "call_to_action",
+            "certificate",
+            "currency",
+            "id",
+            "price",
+            "target_courses",
+            "title",
+            "type",
+        ]
+
+    def get_target_courses(self, product):
+        """
+        For the current product, retrieve its related courses.
+        """
+
+        context = self.context.copy()
+        context.update({"product": product})
+
+        return TargetCourseSerializer(
+            instance=models.Course.objects.filter(
+                product_relations__product=product
+            ).order_by("product_relations__position"),
+            many=True,
+            context=context,
+        ).data
+
+    def get_currency(self, _):  # pylint: disable=no-self-use
+        """
+        Return currency code and symbol
+        to allow frontend to format properly price
+        """
+        (code, symbol) = settings.JOANIE_CURRENCY[:2]
+        return {"code": code, "symbol": symbol}
+
 
 class CourseRunEnrollmentSerializer(serializers.ModelSerializer):
     """
     Enrollment for course run serializer
     """
 
+    id = serializers.CharField(source="uid", read_only=True, required=False)
     resource_link = serializers.CharField(source="course_run.resource_link")
     title = serializers.CharField(source="course_run.title")
-    start = serializers.CharField(source="course_run.start")
-    end = serializers.CharField(source="course_run.end")
-    enrollment_start = serializers.CharField(source="course_run.enrollment_start")
-    enrollment_end = serializers.CharField(source="course_run.enrollment_end")
-    position = serializers.SerializerMethodField()
+    start = serializers.DateTimeField(source="course_run.start")
+    end = serializers.DateTimeField(source="course_run.end")
+    enrollment_start = serializers.DateTimeField(source="course_run.enrollment_start")
+    enrollment_end = serializers.DateTimeField(source="course_run.enrollment_end")
```

Comment:
I'm afraid this allows a user to modify the related course run's fields :scream_cat: 

And actually it's the same for all the fields that we nest under the course: is everything absolutely readonly and do we have tests to ensure it?

Reply:
It seems that DRF `.create()` and `.update()` methods do not support writable nested fields by default.
So unless, these methods has been explicitly overridden, there is no risk to update nested course serializer nested objects.

Nevertheless, I'm updating serializers to set as `read_only` fields by default and write tests on serializer involved into CUD routes.